utt-replica/signature-processor/tests/sigProcessorTests.cpp

```diff
@@ -0,0 +1,519 @@
+  auto dkg = RandSigDKG(thresh, n, Params::NumMessages);
+  auto rc = RegAuthSK::generateKeyAndShares(thresh, n);
+  GpData gp_data{dkg.getCK(), rc.ck_reg};
+  UTTParams d = UTTParams::create((void*)(&gp_data));
+  rc.setIBEParams(d.getParams().ibe);
+  return {d, dkg, rc};
+}
+std::vector<std::shared_ptr<Registrator>> GenerateRegistrators(size_t n, const RegAuthSK& rsk) {
+  std::vector<std::shared_ptr<Registrator>> registrators;
+  std::map<uint16_t, std::string> validation_keys;
+  for (size_t i = 0; i < n; i++) {
+    auto& sk = rsk.shares[i];
+    validation_keys[(uint16_t)i] = serialize<RegAuthSharePK>(sk.toPK());
+  }
+  for (size_t i = 0; i < n; i++) {
+    registrators.push_back(std::make_shared<Registrator>(
+        (uint16_t)i, serialize<RegAuthShareSK>(rsk.shares[i]), validation_keys, serialize<RegAuthPK>(rsk.toPK())));
+  }
+  return registrators;
+}
+
+std::vector<std::shared_ptr<CoinsSigner>> GenerateCommitters(size_t n, const RandSigDKG& dkg, const RegAuthPK& rvk) {
+  std::vector<std::shared_ptr<CoinsSigner>> banks;
+  std::map<uint16_t, std::string> share_verification_keys_;
+  for (size_t i = 0; i < n; i++) {
+    share_verification_keys_[(uint16_t)i] = serialize<RandSigSharePK>(dkg.skShares[i].toPK());
+  }
+  for (size_t i = 0; i < n; i++) {
+    banks.push_back(std::make_shared<CoinsSigner>((uint16_t)i,
+                                                  serialize<RandSigShareSK>(dkg.skShares[i]),
+                                                  serialize<RandSigPK>(dkg.getPK()),
+                                                  share_verification_keys_,
+                                                  serialize<RegAuthPK>(rvk)));
+  }
+  return banks;
+}
+
+std::vector<Client> GenerateClients(size_t c, const RandSigPK& bvk, const RegAuthPK& rvk, const RegAuthSK& rsk) {
+  std::vector<Client> clients;
+  std::string bpk = libutt::serialize<libutt::RandSigPK>(bvk);
+  std::string rpk = libutt::serialize<libutt::RegAuthPK>(rvk);
+
+  for (size_t i = 0; i < c; i++) {
+    std::string pid = "client_" + std::to_string(i);
+    auto sk = rsk.msk.deriveEncSK(rsk.p, pid);
+    auto mpk = rsk.toPK().mpk;
+    std::string csk = libutt::serialize<libutt::IBE::EncSK>(sk);
+    std::string cmpk = libutt::serialize<libutt::IBE::MPK>(mpk);
+    clients.push_back(Client(pid, bpk, rpk, csk, cmpk));
+  }
+  return clients;
+}
+bool cond(uint64_t sigId) {
+  if (full_signatures.empty()) return false;
+  for (const auto& [_, data] : full_signatures) {
+    (void)_;
+    if (data.find(sigId) == data.end()) return false;
+  }
+  return true;
+}
+libutt::api::types::Signature waitAndGetFullSig(uint64_t sigId, uint16_t source) {
+  std::unique_lock lk{sigs_lock};
+  LOG_INFO(logger, "collecting full signature for sid: " << sigId);
+  cv.wait(lk, [&sigId] { return cond(sigId); });
+  return full_signatures.at(source).at(sigId);
+}
+uint16_t n = 4;
+uint16_t t = 3;
+
+int main(int argc, char* argv[]) {
+  // initiate A UTT instance utt1
```

Comment:
Ideally this test should be implemented as a gtest suite? For now we can track it as a tech debt item for faster progress.